test/integration/scenarios/02.basic/verify/manifest_test.go

```diff
@@ -8,22 +8,75 @@ package verify
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"os"
+	"os/exec"
+	"path/filepath"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/model"
 	"github.com/eclipse-symphony/symphony/test/integration/lib/testhelpers"
+	"github.com/princjef/mageutil/shellcmd"
 	"github.com/stretchr/testify/require"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
 )
 
+var (
+	testManifests = []string{
+		"../manifest/oss/solution-container.yaml",
+		"../manifest/oss/target.yaml",
+		"../manifest/oss/solution.yaml",
+		"../manifest/oss/instance.yaml",
+	}
+)
+
+func TestDryRun(t *testing.T) {
+	namespace := os.Getenv("NAMESPACE")
+	if namespace == "" {
+		namespace = "default"
+	}
+	for _, manifest := range testManifests {
+		_, err := DeployManifests(manifest, namespace, "true")
+		require.NoError(t, err)
+	}
+	testBasic_InstanceStatus(t, "0")
+	testBasic_TargetStatus(t, "0")
+	testBasic_VerifyPodsExist(t, []string{}, []string{"nginx", "testapp", namespace + "instance"})
+
+	_, err := DeployManifests("../manifest/oss/target.yaml", namespace, "false")
+	require.NoError(t, err)
+	testBasic_InstanceStatus(t, "0")
```

Comment:
Do we need to check this here? Same as below.

Reply:
Sorry for late reply. I missed the notification email.
We need the check here to check the instance has no deployment if the dryrun is true. Even another reconcile is triggered by target change (dryrun from true to false), instance still has 0 deployment. Things are different if we change dryrun on instance from true to false, the deployment will happen and worker pod will be created.